encoder/vaapiencoder_h264.cpp

```diff
@@ -1142,10 +1326,17 @@ bool  VaapiEncoderH264::pictureReferenceListSet (
 
     for (i = 0; i < m_refList.size(); i++) {
         assert(picture->m_poc != m_refList[i]->m_poc);
-        if (picture->m_poc > m_refList[i]->m_poc)
+        if (picture->m_poc > m_refList[i]->m_poc
```

Comment:
Maybe you misunderstand this code, my main purpose is we can only use reference picture with m_temporalId <= currecnt_picture_temporalId as reference. m_diffPicNumMinus1 is only used let decoder know how to reorder reference list same as encoder's behavior.